webapp/channels/src/components/announcement_bar/notification_permission_bar/index.test.tsx

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {screen, fireEvent, act} from '@testing-library/react';
+import React from 'react';
+
+import {renderWithContext} from 'tests/react_testing_utils';
+import {requestNotificationPermission, isNotificationAPISupported} from 'utils/notifications';
+
+import NotificationPermissionBar from './index';
+
+jest.mock('utils/notifications', () => ({
+    requestNotificationPermission: jest.fn(),
+    isNotificationAPISupported: jest.fn(),
+}));
+
+describe('NotificationPermissionBar', () => {
+    const initialState = {
+        entities: {
+            users: {
+                currentUserId: 'user-id',
+            },
+            general: {
+                config: {},
+                license: {},
+            },
+        },
+    };
+
+    beforeEach(() => {
+        (isNotificationAPISupported as jest.Mock).mockReturnValue(true);
+        (window as any).Notification = {permission: 'default'};
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+        delete (window as any).Notification;
```

Comment:
Is it safe to delete this here when this might be needed for other tests? I think we should probably hold onto whatever was originally stored as `window.Notification` before these tests and then restore it after all of them have run

Reply:
I think its ok to create -> delete -> recreate mocks in test suit, in the above i am running beforeEach as well to create window mock

Reply:
I was thinking it might affect other files. I'm not sure if that's the case though

Reply:
It will only effect in the current test suit